migration-engine/migration-engine-tests/tests/migrations/advisory_locking.rs

```diff
@@ -0,0 +1,207 @@
+use migration_core::{
+    commands::{ApplyMigrationsInput, CreateMigrationInput},
+    GenericApi,
+};
+use std::sync::Arc;
+use tempfile::TempDir;
+use test_macros::test_each_connector;
+use test_setup::{connectors::Tags, TestAPIArgs};
+
+type TestResult = Result<(), anyhow::Error>;
+
+struct TestApi {
+    args: TestAPIArgs,
+    source: String,
+    url: String,
+}
+
+impl TestApi {
+    /// Create a temporary directory to serve as a test migrations directory.
+    pub fn create_migrations_directory(&self) -> anyhow::Result<TempDir> {
+        Ok(tempfile::tempdir()?)
+    }
+
+    async fn new_engine(&self) -> anyhow::Result<Arc<dyn GenericApi>> {
+        Ok(migration_core::migration_api(&self.source).await?)
+    }
+
+    async fn initialize(&self) -> anyhow::Result<()> {
+        if self.args.test_tag.contains(Tags::Postgres) {
+            test_setup::create_postgres_database(&self.url.parse()?).await.unwrap();
+        } else if self.args.test_tag.contains(Tags::Mysql) {
+            test_setup::create_mysql_database(&self.url.parse()?).await.unwrap();
+        } else if self.args.test_tag.contains(Tags::Mssql) {
+            test_setup::create_mssql_database(&self.url).await.unwrap();
+        }
+
+        Ok(())
+    }
+}
+
+async fn mysql_8_test_api(args: TestAPIArgs) -> TestApi {
```

Comment:
I've thought about reducing the verbosity here quite a bit. Ideas are very welcome